### PR TITLE
Add Cover block; Deprecate cover image;

### PR DESCRIPTION
**Pull request by jorgefilipecosta**
## Description
This PR adds a Cover block that is based on the cover image block.
The differences between this PR and the cover image is that this new block uses nesting (simplifying a lot the styles) and this PR allows background videos.

The Cover Image block was deprecated, no automatic migration exists but it should not be possible to insert new Cover Images (the same approach followed on the text columns to columns migration).

To do: transforms.

## How has this been tested?
Verify it is not possible to insert the cover image block.
Add the cover block verify it contains the same functionality offered in the cover image.
Verify now we can use videos as the background.
Verify the video background functionality works correctly on the editor and on the front end.

## Screenshots 
oct-16-2018 00-42-10

### CODE REVIEW

File: packages/block-library/src/cover-image/index.js [hunk 1]
```diff
@@ -23,6 +23,7 @@ import {
 	withColors,
 	getColorClassName,
 } from '@wordpress/editor';
+import deprecated from '@wordpress/deprecated';
 
 const validAlignments = [ 'left', 'center', 'right', 'wide', 'full' ];
```
Comment (ZebulanStanphill) [inline packages/block-library/src/cover-image/index.js]: Shouldn't this constant use CAPS_CASE like the constants in `edit.js`?

Comment (ZebulanStanphill) [reply packages/block-library/src/cover-image/index.js]: Whoops, I just noticed this is the deprecated block. Well, I guess the point still applies, but it isn't really important. :stuck_out_tongue:

**Comment by ZebulanStanphill** [comment]
I'm guessing you already have this planned, but there should be a transform available to transform existing Cover Image blocks to Cover blocks, similar to what was done for the other deprecated blocks (Subheading and Text Columns), so that users can more easily convert the old blocks to the replacements.

Also, what if you could use a background image as a fallback for browsers that didn't support autoplaying videos (e.g. every major mobile browser)?

I'm noticing quite a bit of overlap with #10562. A lot of the features of this block could also exist in a Container block. Actually, I think a fully-featured Container block would do everything this block does and more. That is not to say they can't both exist, but I think at least some of the stuff done here could be copied over into #10562.

**Comment by youknowriad** [comment]
The Cover Image block seems like a block that is used a lot. I wonder if it's the right opportunity to introduce the "cross-block" migration to the deprecated block feature.

**Comment by ZebulanStanphill** [comment]
@youknowriad That certainly seems like something that should be implemented before WP 5.0, in my opinion.

**Comment by jorgefilipecosta** [comment]
I'm closing this PR it was a proof of concept but did too many things making it hard to review.
We will go in smaller steps. The first step is just applying a rename to cover image using an approach similar to what happened in core/text -> core/paragraph. And add video support to it.

**Pull request closed without merging**